pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -170,36 +168,26 @@ func (d *DualWriterMode2) DeleteCollection(ctx context.Context, deleteValidation
 	if err != nil {
 		log.WithValues("deleted", deletedLegacy).Error(err, "failed to delete collection successfully from legacy storage")
 		d.recordLegacyDuration(true, mode2Str, d.resource, method, startLegacy)
-		return deletedLegacy, err
-	}
-	d.recordLegacyDuration(false, mode2Str, d.resource, method, startLegacy)
-
-	legacyList, err := meta.ExtractList(deletedLegacy)
-	if err != nil {
-		log.Error(err, "unable to extract list from legacy storage")
-		return nil, err
-	}
-
-	// Only the items deleted by the legacy DeleteCollection call are selected for deletion by Storage.
-	_, err = parseList(legacyList)
```

Comment:
I did not understand why we call those both functions, to me they didn't made any sense here. I removed them.